### PR TITLE
empty charset fix

### PR DESCRIPTION
**Pull request by ahorek**
fixes https://github.com/mikel/mail/issues/1299

### CODE REVIEW

**Comment by ahorek** [comment]
the actual problem is in action mailer, that always expects a charset to be defined and https://github.com/mikel/mail/pull/1145 broke it, however I think it should be fixed here, otherwise it'll break older rails versions

@mikel what about add action mailer's tests to travis?

**Comment by jeremy** [comment]
@ahorek, that's a good idea. Do Action Mailer tests fail with #1145? We could test Action Mailer against Mail master branch as well.

**Comment by ahorek** [comment]
> Do Action Mailer tests fail with #1145?

yes, they do fail. I'm working on CI integration in #1303

**Comment by jeremy** [comment]
Great! Let's get a PR going in Rails as well. We can track Mail master branch in an additional CI job there.

File: lib/mail/message.rb [hunk 1]
```diff
@@ -1503,8 +1503,10 @@ def charset
     # Sets the charset to the supplied value.
     def charset=(value)
       @defaulted_charset = false
-      @charset = value
-      @header.charset = value
+      if value
+        @charset = value
+        @header.charset = value
+      end
```
Comment (jeremy) [inline lib/mail/message.rb]: How would someone remove the charset now, to rely on default parsing?

This is seeming like something to fix downstream in Action Mailer.

Comment (ahorek) [reply lib/mail/message.rb]: it isn't just "bad tests", it breaks Action Mailer for real. Even if it'll be fixed in Action Mailer, Mail 2.8 will be incompatible with all previous Rails versions, I think we should avoid that for all costs.

what about skipping blank parameters? https://github.com/mikel/mail/blob/master/lib/mail/fields/parameter_hash.rb#L47

Comment (ahorek) [reply lib/mail/message.rb]: ok, I fixed that by removing the existing header if you set charset = nil

note that this case didn't work even on previous mail versions including 2.7.1 stable

**Comment by ahorek** [comment]
if you have time, please review https://github.com/mikel/mail/pull/1308 https://github.com/mikel/mail/pull/1301 these two PRs are preventing a green build on travis

**Comment by ahorek** [comment]
ping @jeremy 

it should be ready, what do you think? any objections? travis build is still failing because of https://github.com/mikel/mail/pull/1308

**Pull request merged**